Jetsnack/app/src/main/java/com/example/jetsnack/ui/home/cart/Cart.kt

```diff
@@ -167,12 +161,100 @@ private fun CartContent(
-                onSnackClick = onSnackClick
+            SwipeDismissItem(
+                background = { offsetX ->
+                    val paddingRight = when {
+                        offsetX == 0f || offsetX < -300f -> 0.dp
+                        (offsetX < -10f && offsetX > -300f) -> ((offsetX * -24) / 350).toInt().dp
+
+                        else -> 24.dp
+                    }
+                    val background = if (offsetX >= -450) {
+                        Color.Transparent
+                    } else {
+                        JetsnackTheme.colors.error
+                    }
+                    Row(
+                        modifier = Modifier
+                            .fillMaxWidth()
+                            .heightIn(120.dp, 120.dp)
+                            .padding(end = paddingRight)
+                            .background(background),
+                        horizontalArrangement = Arrangement.End,
+                        verticalAlignment = Alignment.CenterVertically
+
+                    ) {
+                        val size = when {
+                            offsetX == 0f -> 0.dp
+                            (offsetX < -0f) -> (offsetX / -5).dp
+                            (offsetX < -200f) -> (offsetX * -10).dp
+                            else -> 0.dp
+                        }
+                        val sizeModifier = if (offsetX > -300f) {
+                            Modifier.size(size)
+                        } else {
+                            Modifier
+                                .height(size * 1.8f)
+                                .width(size * 1.8f)
+                        }
+                        val shape = if (offsetX >= -300) CircleShape else RoundedCornerShape(30.dp)
+                        Box(
+                            modifier = Modifier
+                                .clip(shape)
+                                .background(JetsnackTheme.colors.error)
+                                .then(sizeModifier)
+                        ) {
+                            Row(
+                                modifier = Modifier
+                                    .fillMaxWidth()
+                                    .fillMaxHeight(),
+                                horizontalArrangement = Arrangement.Center,
+                                verticalAlignment = Alignment.CenterVertically
+                            ) {
+                                if (offsetX < -150f && offsetX > -450) {
+                                    val iconAlpha = if (offsetX < -250) 0.5f else 1f
+                                    Icon(
+                                        imageVector = Icons.Filled.DeleteForever,
+                                        modifier = Modifier
+                                            .width(16.dp)
+                                            .height(18.dp),
+                                        tint = JetsnackTheme.colors.uiBackground
```

Comment:
Again, use `Modifier.alpha` (which uses `Modifier.graphicsLayer`) on the `Icon` to change the color and only require re-draw.

Reply:
Looks like graphicsLayer is being used now